lib/internal/process/promises.js

```diff
@@ -26,6 +33,7 @@ function setupPromises(scheduleMicrotasks) {
     if (hasBeenNotified !== undefined) {
       hasBeenNotifiedProperty.delete(promise);
       if (hasBeenNotified === true) {
+        promiseInternals.untrackPromise(promise);
```

Comment:
Does this happen for every rejected promise? It looks like this will create a `Persistent` handle for each one... Not sure if its good for GC performance cc @nodejs/v8 @trevnorris